config/zabbix-agent-lts/zabbix-agent-lts.inc

```diff
@@ -257,11 +257,28 @@ EOF;
 						"stop" => "$zagent_stop"
 				)
 		);
-		mwexec("{$zagent_rcfile} restart");	
+
+		// Check if zabbix_agentd is currently running
+		// if it's running, is necessary restart the service
+		// because there are not a HUP signal.
+		if (!is_process_running("zabbix_agentd")) {
+			mwexec("{$zagent_rcfile} start");
+		}else{
+			mwexec("{$zagent_rcfile} stop");
+			// Check if zabbix_agentd is still running
+			if (is_process_running("zabbix_agentd")) {
+				sigkillbyname("zabbix_agentd", "KILL");
+				sleep(2);
+				unlink_if_exists("/var/run/zabbix-agent-lts/zabbix_agent_lts.pid");
+			}
+			// Start zabbix_agentd
+			mwexec("{$zagent_rcfile} start");
```

Comment:
This dirty game (kill -TERM, check, sleep, kill -KILL, ...) should be done inside rc script instead of at PHP side. It should work as expected since stop_service() waits until rc file is executed.